src/pages/profile/verify.rs

```diff
@@ -57,42 +72,71 @@ pub fn Verify(cx: Scope, id: String) -> Element {
+    let task_verify_to_device = task_verify.clone();
+
+    use_coroutine(cx, |mut _rx: UnboundedReceiver<()>| {
+        to_owned![task_wait_confirmation, client, task_verify, task_handle_error, task_handle_error_a, task_handle_error_b, task_handle_error_c, task_handle_error_d];
 
         async move {
             client.add_event_handler(
-                |ev: ToDeviceKeyVerificationRequestEvent, client: Client| async move {
+                move |ev: ToDeviceKeyVerificationRequestEvent, client: Client| {
+                    let task_handle_error_a = task_handle_error_a.clone();
+                    async move {
                     info!("here ToDeviceKeyVerificationRequestEvent");
                     let request = client
                         .encryption()
                         .get_verification_request(&ev.sender, &ev.content.transaction_id)
                         .await
                         .expect("Request object wasn't created");
 
-                    request
-                        .accept()
-                        .await
-                        .expect("Can't accept verification request");
-                },
+                    if let Err(_) = request.accept().await {
+                        task_handle_error_a.send(VerificationError::SasAcceptFailed)
+                    };
+                }},
             );
 
             client.add_event_handler(
-                |ev: ToDeviceKeyVerificationStartEvent, client: Client| async move {
-                    if let Some(Verification::SasV1(sas)) = client
-                        .encryption()
-                        .get_verification(&ev.sender, ev.content.transaction_id.as_str())
-                        .await
-                    {
-                        info!(
-                            "ToDeviceKeyVerificationStartEvent Starting verification with {} {}",
-                            &sas.other_device().user_id(),
-                            &sas.other_device().device_id()
-                        );
-                        // print_devices(&ev.sender, &client).await;
-                        let _ = sas.accept().await;
+                move |ev: ToDeviceKeyVerificationStartEvent, client: Client| {
+                    let task_handle_error_b = task_handle_error_b.clone();
+                    async move {
+                        if let Some(Verification::SasV1(sas)) = client
+                            .encryption()
+                            .get_verification(&ev.sender, ev.content.transaction_id.as_str())
+                            .await
+                        {
+                            info!(
+                                "ToDeviceKeyVerificationStartEvent Starting verification with {} {}",
+                                &sas.other_device().user_id(),
+                                &sas.other_device().device_id()
+                            );
+                            // print_devices(&ev.sender, &client).await;
```

Comment:
Comments can easily go stale or be forgotten, either we remove them or do a debug!/trace! log that can be disabled for production builds
```suggestion
```